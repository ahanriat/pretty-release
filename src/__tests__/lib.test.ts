import { ugglyReleaseNote, ugglyReleaseNote2 } from './blob/release-note-1';
import { parseRelease, prettifyRelease } from '~/lib';

describe('pretty release', () => {
  describe('parseRelease', () => {
    test('Parse the messages', () => {
      expect(parseRelease(ugglyReleaseNote).messages.length).toBe(32);
      expect(parseRelease(ugglyReleaseNote).messages[0]).toEqual(
        '- 🐛Fix missing translation on WorkoutPlanReadyScreen (#1891) (Antoine Hanriat)',
      );
    });
    test('Parse the messages 2/', () => {
      expect(parseRelease(ugglyReleaseNote2).messages).toEqual([
        '- Fix/Double workouts logged (#2976) (Nikita Nikitin)',
        '- Fix Env used in building iOS app (#3018) (Mohamed Shaban)',
      ]);
    });
  });

  describe('prettifyRelease', () => {
    test('Prettify the release note', () => {
      expect(prettifyRelease(ugglyReleaseNote)).toMatchSnapshot('prettyRelease');
    });
    test('Prettify the release note 2/', () => {
      expect(prettifyRelease(ugglyReleaseNote2)).toMatchSnapshot('prettyRelease2');
    });
    test('Prettify the release note is re entrant', () => {
      expect(prettifyRelease(prettifyRelease(ugglyReleaseNote))).toMatchSnapshot('prettyRelease');
    });
  });
});
