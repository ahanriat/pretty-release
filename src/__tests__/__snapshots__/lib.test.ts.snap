// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pretty release prettifyRelease Prettify the release note 2/: prettyRelease2 1`] = `
"
### Features 💎:



### Enhancements ⚡️:



### Analytics 📊:



### Library upgrade ⬆️:



### Technical 👨‍🔧:



### Tooling 🛠:



### Bug fixes 🐛:
  - Fix Env used in building iOS app (#3018) (Mohamed Shaban)
  - Fix/Double workouts logged (#2976) (Nikita Nikitin)


### Doc 📖:



### To be sorted 👈:
"
`;

exports[`pretty release prettifyRelease Prettify the release note is re entrant: prettyRelease 1`] = `
"
### Features 💎:
  - Experiment: Hard conclude 1803-currency-symbols (#1887) (Antoine Hanriat)
  - 🚛 Feat: Exercise store (#1898) (Stefano Bourscheid)


### Enhancements ⚡️:
  - DS-56 💄Improvement/segmented slider (#1722) (Arkalyk Akash)
  - Enhancement/add screen context snowplow (#1913) (alex-blair)
  - 💄Center text on ChnageStartsHereScene (#1906) (Antoine Hanriat)


### Analytics 📊:
  - Add Snowplow events (#1882) (alex-blair)


### Library upgrade ⬆️:
  - ⬆Upgrade appboy-sdk (Braze) (#1950) (Antoine Hanriat)
  - ⬆️Upgrade React-native to 57.5 and React to 16.6 (#1819) (Antoine Hanriat)
  - 🌈Upgrade and fix Storybook (#1916) (Antoine Hanriat)


### Technical 👨‍🔧:
  - Chore/loading states (#1919) (Nikita Nikitin)


### Tooling 🛠:



### Bug fixes 🐛:
  - BUG-165 Profile | Weight difference value is reset to 0 after login (#1902) (Arkalyk Akash)
  - Fix Readme links to react/ -> src/ (ahanriat)
  - Fix static libs 3rd try [TI-79] 😅 (#1900) (Desmond McNamee)
  - Fix/BUG 144 Email address display (#1828) (Nikita Nikitin)
  - Fixed overlapping text on Rest Screen during Cooldown (#1872) (Carly Lee)
  - 🐛 Fix crash when using SVG component (#1907) (Desmond McNamee)
  - 🐛Fix cropped accents (#1894) (Antoine Hanriat)
  - 🐛Fix date format for workout schedule screen (#1895) (Antoine Hanriat)
  - 🐛Fix floatingBottomDecorator pointer events to allow access elements behind such as checkbox on GDPR screen (#1890) (Antoine Hanriat)
  - 🐛Fix missing translation on WorkoutPlanReadyScreen (#1891) (Antoine Hanriat)
  - 🐛Fix newsletter enrolment (#1888) (Antoine Hanriat)


### Doc 📖:
  - Few words on the branches in wiki (#1878) (Nikita Nikitin)
  - Wiki update for Detox (#1920) (Steve Chun)


### To be sorted 👈:
  - Automated releases for GitHub (#1901) (Steve Chun)
  - Detox - enable settings and today tab test suites for iOS (#1868) (Steve Chun)
  - DownloadError component for migration  (#1893) (Carly Lee)
  - MT-585 Update new Snapshot screen to display approximations (#1827) (Arkalyk Akash)
  - Remove deprecated PlansScreen (#1918) (alex-blair)
  - Save body fat value onScroll (#1899) (alex-blair)
  - Update post-workout kcal estimation (#1932) (alex-blair)
  - Use legacy build system for testing builds (#1912) (Steve Chun)
  - [WKT-499][TI-74] WorkoutTemplate API migration (#1816) (Carly Lee)"
`;

exports[`pretty release prettifyRelease Prettify the release note: prettyRelease 1`] = `
"
### Features 💎:
  - Experiment: Hard conclude 1803-currency-symbols (#1887) (Antoine Hanriat)
  - 🚛 Feat: Exercise store (#1898) (Stefano Bourscheid)


### Enhancements ⚡️:
  - DS-56 💄Improvement/segmented slider (#1722) (Arkalyk Akash)
  - Enhancement/add screen context snowplow (#1913) (alex-blair)
  - 💄Center text on ChnageStartsHereScene (#1906) (Antoine Hanriat)


### Analytics 📊:
  - Add Snowplow events (#1882) (alex-blair)


### Library upgrade ⬆️:
  - ⬆Upgrade appboy-sdk (Braze) (#1950) (Antoine Hanriat)
  - ⬆️Upgrade React-native to 57.5 and React to 16.6 (#1819) (Antoine Hanriat)
  - 🌈Upgrade and fix Storybook (#1916) (Antoine Hanriat)


### Technical 👨‍🔧:
  - Chore/loading states (#1919) (Nikita Nikitin)


### Tooling 🛠:



### Bug fixes 🐛:
  - BUG-165 Profile | Weight difference value is reset to 0 after login (#1902) (Arkalyk Akash)
  - Fix Readme links to react/ -> src/ (ahanriat)
  - Fix static libs 3rd try [TI-79] 😅 (#1900) (Desmond McNamee)
  - Fix/BUG 144 Email address display (#1828) (Nikita Nikitin)
  - Fixed overlapping text on Rest Screen during Cooldown (#1872) (Carly Lee)
  - 🐛 Fix crash when using SVG component (#1907) (Desmond McNamee)
  - 🐛Fix cropped accents (#1894) (Antoine Hanriat)
  - 🐛Fix date format for workout schedule screen (#1895) (Antoine Hanriat)
  - 🐛Fix floatingBottomDecorator pointer events to allow access elements behind such as checkbox on GDPR screen (#1890) (Antoine Hanriat)
  - 🐛Fix missing translation on WorkoutPlanReadyScreen (#1891) (Antoine Hanriat)
  - 🐛Fix newsletter enrolment (#1888) (Antoine Hanriat)


### Doc 📖:
  - Few words on the branches in wiki (#1878) (Nikita Nikitin)
  - Wiki update for Detox (#1920) (Steve Chun)


### To be sorted 👈:
  - Automated releases for GitHub (#1901) (Steve Chun)
  - Detox - enable settings and today tab test suites for iOS (#1868) (Steve Chun)
  - DownloadError component for migration  (#1893) (Carly Lee)
  - MT-585 Update new Snapshot screen to display approximations (#1827) (Arkalyk Akash)
  - Remove deprecated PlansScreen (#1918) (alex-blair)
  - Save body fat value onScroll (#1899) (alex-blair)
  - Update post-workout kcal estimation (#1932) (alex-blair)
  - Use legacy build system for testing builds (#1912) (Steve Chun)
  - [WKT-499][TI-74] WorkoutTemplate API migration (#1816) (Carly Lee)"
`;
